### PR TITLE
Add basic black-holing information to guards.

### PR DESCRIPTION
**Pull request by vext01**
We record the position of the guard in SIR and the live TIR variables at the point of the guard.

### CODE REVIEW

File: yktrace/src/tir.rs [hunk 1]
```diff
@@ -551,13 +574,33 @@ impl Display for TirTrace<'_> {
...
     /// The requirement upon `val` for the guard to pass.
-    pub kind: GuardKind
+    pub kind: GuardKind,
+    /// The block whose terminator was the basis for this guard. This is here so that, in the event
+    /// that the guard fails, we know where to start the blackhole interpreter.
+    pub block: GuardBlock,
+    /// The TIR locals that are live at the time of the guard (in addition to the trace I/O
+    /// variable, $1, which is assumed to be live throughout the entirety of the trace). This is
+    /// needed so that we can initialise the blackhole interpreter with the correct state before
+    /// starting it executing.
```
Comment (ptersilie) [inline yktrace/src/tir.rs:602]: ...before executing it.

**Comment by ptersilie** [comment]
Just one minor comment.

**Comment by vext01** [comment]
Give that a shot.

**Comment by ptersilie** [comment]
Please squash.

**Comment by vext01** [comment]
splat.

**Pull request merged**